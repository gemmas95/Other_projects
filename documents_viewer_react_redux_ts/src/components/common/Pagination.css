.paginationList {
  display: flex;
  list-style-type: none;
  justify-content: center;
  margin-top: auto;
}

.paginationList > li button {
  border: 1px solid #dee2e6;
  margin: 5px;
  color: #39dc73;
  font-weight: bold;
}

.paginationList > li a,
.paginationList > li button {
  /* padding: 15px 20px; */
  padding: 7px 13px;
  cursor: pointer;
  font-size: 17px;
}
.paginationList button:hover {
  color: white;
  background-color: #39dc73;
}
.paginationList button:focus {
  z-index: 2;
  outline: 0;
  background-color: #39dc73;
  color: white;
  box-shadow: 0 0 0 0.2rem rgba(35, 77, 87, 0.2);
}

.paginationList > li:focus {
  color: white;
  background-color: purple;
}
