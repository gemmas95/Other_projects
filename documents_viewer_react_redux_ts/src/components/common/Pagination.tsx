import React from "react";
import "./Pagination.css";

interface Props {
  documentsPerPage: number;
  totalDocuments: number;
  paginate: Function;
}

const Pagination = ({
  documentsPerPage,
  totalDocuments,
  paginate,
}: Props): JSX.Element => {
  const pageNumbers = [];

  for (let i = 1; i <= Math.ceil(totalDocuments / documentsPerPage); i++) {
    pageNumbers.push(i);
  }

  return (
    <nav>
      <ul className="paginationList">
        {pageNumbers.map((number: number) => (
          <li key={number} className="page-item">
            <button onClick={() => paginate(number)}>{number}</button>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Pagination;
