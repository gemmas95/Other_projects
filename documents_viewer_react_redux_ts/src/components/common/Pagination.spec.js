import renderer from "react-test-renderer";
import React from "react";
import Pagination from "./Pagination";

describe("Pagination", () => {
  it("should match snapshot without data", () => {
    const PaginationTree = renderer.create(<Pagination />);
    expect(PaginationTree.toJSON()).toMatchSnapshot();
  });
});
