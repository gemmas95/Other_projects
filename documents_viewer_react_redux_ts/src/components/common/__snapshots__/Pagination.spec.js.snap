// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match snapshot without data 1`] = `
<nav>
  <ul
    className="paginationList"
  />
</nav>
`;
